pkg/tenant/tenants_test.go

```diff
@@ -85,6 +85,42 @@ func (suite *tenantTestSuite) TestTenantDetail() {
 	require.Equal(req.EnvironmentType, reply.EnvironmentType, "tenant environment type should match")
 }
 
+func (suite *tenantTestSuite) TestTenantUpdate() {
+	require := suite.Require()
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+
+	defer cancel()
+
+	// Connect to mock trtl database
+	trtl := db.GetMock()
+	defer trtl.Reset()
+
+	// Call the OnPut method and return a PutReply.
+	trtl.OnPut = func(ctx context.Context, pr *pb.PutRequest) (*pb.PutReply, error) {
+		return &pb.PutReply{}, nil
+	}
+
+	// Should return an error if the tenant name does not exist
+	_, err := suite.client.TenantUpdate(ctx, &api.Tenant{ID: "01ARZ3NDEKTSV4RRFFQ69G5FAV", EnvironmentType: "prod"})
+	suite.requireError(err, http.StatusBadRequest, "tenant name is required", "expected error when tenant name does not exist")
+
+	// Should return an error if the tenant environment type does not exist
+	_, err = suite.client.TenantUpdate(ctx, &api.Tenant{ID: "01ARZ3NDEKTSV4RRFFQ69G5FAV", Name: "example-dev"})
+	suite.requireError(err, http.StatusBadRequest, "tenant environment type is required", "expected error when tenant environent type does not exist")
+
+	req := &api.Tenant{
+		ID:              "01ARZ3NDEKTSV4RRFFQ69G5FAV",
```

Comment:
Can we change this to a different ID so that we're testing that the ID in the request does not override the ID on the tenant object?